pkg/controller/core/workload_controller.go

```diff
@@ -193,7 +193,7 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		if updated {
 			return ctrl.Result{}, workload.ApplyAdmissionStatus(ctx, r.client, &wl, true)
 		}
-	} else if !workload.IsEvictedByDeactivation(&wl) {
+	} else if cond := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadEvicted); cond == nil || cond.Status == metav1.ConditionFalse {
```

Comment:
use this instead `!apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadEvicted)`

Reply:
That is a good point. Thank you.